packages/react/src/collapsible/panel/useCollapsiblePanel.ts

```diff
@@ -120,19 +144,31 @@ export function useCollapsiblePanel(
 
     if (element) {
       const isBeforeMatch = isBeforeMatchRef.current;
-      const isInitiallyOpen = isInitialOpenRef.current;
+      const isInitiallyOpen = isInitiallyOpenRef.current;
+      const isInitialOpenAnimation = isInitialOpenAnimationRef.current;
       const isTransitioning = isTransitioningRef.current;
+
       const originalAnimationName =
         element.style.animationName === 'none' ? '' : element.style.animationName;
       const originalTransitionDuration = originalTransitionDurationStyleRef.current;
+
       // cancel animation/transitions for these specific instances:
-      // 1. when initially open, on mount/load, it should just appear fully open but remain animated per styles afterwards
-      // 2. when using `hidden='until-found'` and is opened by find-in-page, it should open instantly but remain animated //    as styled afterwards
-      const shouldCancelAnimation = isBeforeMatch || isInitiallyOpen;
+      // 1. when initially open, on mount/load, it should just appear fully open
+      //    but remain animated per styles afterwards
+      // 2. when using `hidden='until-found'` and is opened by find-in-page, it
+      //    should open instantly but remain animated //    as styled afterwards
+      const shouldCancelAnimation = isBeforeMatch || isInitialOpenAnimation;
 
       element.style.animationName = 'none';
 
-      const isClosed = !open && !contextMounted;
+      const isClosed = !open && !mounted;
+
+      if (!keepMounted && !isInitiallyOpen) {
+        // when keepMounted is false and the initial state is closed, the panel
+        // does not exist in the DOM so transition listeners were never registered
+        // they need to be manually registered before any state changes
+        registerCssTransitionListeners();
+      }
```

Comment:
This was the biggest gotcha for adding `keepMounted={false}`, there is no way to directly check if listeners have been registered or not